packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -77,10 +85,42 @@ def __init__(
 
         res = MPCTensor._mpc_from_shares(shares, parties)
 
+        self.parties = parties
+
         self.mpc_shape = shape
 
+        # we need to make sure that when we zip up clients from
+        # multiple MPC tensors that they are in corresponding order
+        # so we always sort all of them by the id of the domain
+        # TODO: store children as lists of dictionaries because eventually
+        # it's likely that we have multiple shares from the same client
+        # (For example, if you wanted a domain to have 90% share ownership
+        # you'd need to produce 10 shares and give 9 of them to the same domain)
+        res.sort(key=lambda share: share.client.name + share.client.id.no_dash)
+
         super().__init__(res)
 
+    def publish(self, sigma: float) -> MPCTensor:
+        """Note: @George has proposed that we could just add .publish() to the
```

Comment:
Should this be added to the backlog and comment removed?

Reply:
nah just removing it for now